_posts/2020-07-21-cube-functionality-to-kartothek.markdown

```diff
@@ -0,0 +1,410 @@
+...     dimension_columns=["city", "day"],
+...     partition_columns=["country"]
+...)
+```
+
+We use the simple **kartothek.io.eager_cube** backend to store the data:
+
+```python
+>>> from kartothek.io.eager_cube import build_cube
+>>> datasets_build = build_cube(
+...   data=df_weather,
+...   store=store,
+...   cube=cube
+...)
+```
+
+
+where **store** is the **simplekv** store of storefactory. (For more details, please refer our [Kartothek](2019-05-28-introducing-kartothek.markdown) post.)
+
+We have just preserved a single Kartothek dataset. Let's print the content of seed dataset.
+
+```python
+>>> print(", ".join(sorted(datasets_build.keys())))
+seed
+>>> ds_seed = datasets_build["seed"].load_all_indices(store)
+>>> print(ds_seed.uuid)
+geodata++seed
+>>> print(", ".join(sorted(ds_seed.indices)))
+city, country, day
+```
+
+Now we have a quick look at the store content. Note that we cut out UUIDs and timestamps here for brevity:
+
+```python
+>>> import re
+>>> def print_filetree(s, prefix=""):
+...     entries = []
+...     for k in sorted(s.keys(prefix)):
+...         k = re.sub("[a-z0-9]{32}", "<uuid>", k)
+...         k = re.sub("[0-9]{4}-[0-9]{2}-[0-9]{2}((%20)|(T))[0-9]{2}%3A[0-9]{2}%3A[0-9]+.[0-9]{6}", "<ts>", k)
+...         entries.append(k)
+...     print("\n".join(sorted(entries)))
+>>> print_filetree(store)
+geodata++seed.by-dataset-metadata.json
+geodata++seed/indices/city/<ts>.by-dataset-index.parquet
+geodata++seed/indices/day/<ts>.by-dataset-index.parquet
+geodata++seed/table/_common_metadata
+geodata++seed/table/country=DE/<uuid>.parquet
+geodata++seed/table/country=UK/<uuid>.parquet
+```
+
+We can **extend** this cube by adding new columns to the dataframes.
+
+## Extend Operation
+
+Now let’s say we also would like to have longitude and latitude data in our cube.
+
+```python
+>>> from kartothek.io.eager_cube import extend_cube
+>>> df_location = pd.read_csv(
+...     filepath_or_buffer=StringIO("""
+...    city country  latitude  longitude
+... Hamburg      DE 53.551086   9.993682
+... Dresden      DE 51.050407  13.737262
+...  London      UK 51.509865  -0.118092
+...   Tokyo      JP 35.652832 139.839478
+...     """.strip()),
+...     delim_whitespace=True,
+... )
+```
+
+```python
+>>> datasets_extend = extend_cube(
+...   data={"latlong": df_location},
+...   store=store,
+...   cube=cube,
+... )
+```
+
+This results in an extra dataset:
+
+```python
+>>> print(", ".join(sorted(datasets_extend.keys())))
+latlong
+>>> ds_latlong = datasets_extend["latlong"].load_all_indices(store)
+>>> print(ds_latlong.uuid)
+geodata++latlong
+>>> print(", ".join(sorted(ds_latlong.indices)))
+country
+```
+Note that for the second dataset, no indices for **city** and **day** exist. 
+These are only created for the seed dataset, since that dataset forms the groundtruth about which `city-day` entries are part of the cube.
+(Dataset that provides the groundtruth about which Cells are in the cube is called the **seed dataset**).
```

Comment:
I would just delete the sentence in l. 162, it is somehow redundant

Reply:
have removed above line.